donut/modules/uploads/helpers.py

```diff
@@ -0,0 +1,91 @@
+import flask
+import os
+import glob
+from donut.modules.uploads.upload_permission import UploadPermissions
+from donut.auth_utils import check_permission, check_login
+
+ALLOWED_EXTENSIONS = set(
+    ['docx', 'doc', 'txt', 'pdf', 'png', 'jpg', 'jpeg', 'gif'])
+
+
+def read_page(url):
+    '''
+    Reads in content from a markdown file
+    '''
+    root = os.path.join(flask.current_app.root_path,
+                        flask.current_app.config["UPLOAD_WEBPAGES"])
+    path = os.path.join(root, url + '.md')
+    if not os.path.isfile(path):
+        return None
+    with open(path) as f:
+        return f.read()
+
+
+def allowed_file(filename):
+    '''
+    Checks for allowed file extensions.
+    '''
+    return '.' in filename and \
+           filename.rsplit('.', 1)[1].lower() in ALLOWED_EXTENSIONS
+
+
+def remove_link(filename):
```

Comment:
Can this be combined with the similar function in `editor`? Similarly for `read_page()`

Reply:
similar but not similar enough imo...the replace strings are different and probably a hassle to deal with